drivers/clock_control/clock_fixed_rate.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (C) 2022 Google, LLC
+ *
+ */
+
+#include <zephyr/drivers/clock_control.h>
+
+#define DT_DRV_COMPAT fixed_clock
+
+struct fixed_rate_clock_config {
+	const struct device *clk_dev;
+	clock_control_subsys_t clk_subsys;
+	uint32_t rate;
+};
+
+static int fixed_rate_clk_init(const struct device *dev)
+{
+	const struct fixed_rate_clock_config *config = dev->config;
+
+	if (!config->clk_dev)
+		return 0;
+
+	return clock_control_on(config->clk_dev, config->clk_subsys);
+}
+
+static int fixed_rate_clk_on(const struct device *dev,
+			     clock_control_subsys_t sys)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(sys);
+
+	return 0;
+}
+
+static int fixed_rate_clk_off(const struct device *dev,
+			      clock_control_subsys_t sys)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(sys);
+
+	return 0;
+}
+
+static enum clock_control_status fixed_rate_clk_get_status(const struct device *dev,
+							   clock_control_subsys_t sys)
+{
+	return CLOCK_CONTROL_STATUS_ON;
+}
+
+static int fixed_rate_clk_get_rate(const struct device *dev,
+				   clock_control_subsys_t sys,
+				   uint32_t *rate)
+{
+	const struct fixed_rate_clock_config *config = dev->config;
+
+	ARG_UNUSED(sys);
+
+	*rate = config->rate;
```

Comment:
I would expect that if this clock has no factor and fully depends from a known clock, then its rate also depend from the input clock.
Am I missing something ?

Reply:
That would be a `compatible = "fixed-factor-clock"` device. Perhaps both types of fixed clocks could be handled by the same driver, though?

Reply:
@henrikbrixandersen do you think that's preferable to two separate drivers?

Reply:
@mfischer Depends on the amount of shared code between the two drivers. If most of the code is shared, I think one common driver with support for both compatibles is preferable.

Reply:
The follow up commit to this one has the code for 'fixed-factor-clock'. I'll add tests and see if merging both into one driver buys me much.